src/main/java/mekanism/common/integration/framedblocks/ChemicalModel.java

```diff
@@ -0,0 +1,123 @@
+package mekanism.common.integration.framedblocks;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import com.google.common.base.Preconditions;
+import com.mojang.math.Transformation;
+import mekanism.api.chemical.Chemical;
+import mekanism.common.Mekanism;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.ItemOverrides;
+import net.minecraft.client.renderer.texture.TextureAtlas;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.resources.model.BakedModel;
+import net.minecraft.client.resources.model.ModelBakery;
+import net.minecraft.client.resources.model.ModelResourceLocation;
+import net.minecraft.client.resources.model.ModelState;
+import net.minecraft.client.resources.model.UnbakedModel;
+import net.minecraft.core.Direction;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.neoforged.neoforge.client.ChunkRenderTypeSet;
+import net.neoforged.neoforge.client.model.SimpleModelState;
+import net.neoforged.neoforge.client.model.data.ModelData;
+import org.jetbrains.annotations.Nullable;
+
+final class ChemicalModel implements BakedModel {
+
+    private static final ModelState SIMPLE_STATE = new SimpleModelState(Transformation.identity());
+    static final ModelResourceLocation BARE_MODEL = ModelResourceLocation.standalone(FramedBlocksIntegration.Constants.CHEMICAL_DUMMY_MODEL);
+
+    private final RenderType renderType;
+    private final ChunkRenderTypeSet renderTypeSet;
+    private final Map<Direction, List<BakedQuad>> quads;
+    private final TextureAtlasSprite particles;
+
+    private ChemicalModel(RenderType renderType, Map<Direction, List<BakedQuad>> quads, TextureAtlasSprite particles) {
+        this.renderType = renderType;
+        this.renderTypeSet = ChunkRenderTypeSet.of(renderType);
+        this.quads = quads;
+        this.particles = particles;
+    }
+
+    @Override
+    public List<BakedQuad> getQuads(@Nullable BlockState state, @Nullable Direction side, RandomSource random) {
+        return getQuads(state, side, random, ModelData.EMPTY, RenderType.translucent());
+    }
+
+    @Override
+    public List<BakedQuad> getQuads(@Nullable BlockState state, @Nullable Direction side, RandomSource rand, ModelData extraData, RenderType layer) {
+        return (side == null || layer != renderType) ? List.of() : quads.get(side);
```

Comment:
This should return `Collections.emptyList()` instead of `List.of()` as the backing implementation is more efficient in a variety of cases due to it short circuiting various methods instead of just basically being an immutable view of an array list.